src/main/java/com/liveramp/captain/example/ExampleCaptainWorkflow.java

```diff
@@ -0,0 +1,349 @@
+package com.liveramp.captain.example;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.liveramp.captain.app_type.CaptainAppType;
+import com.liveramp.captain.daemon.CaptainBuilder;
+import com.liveramp.captain.daemon.CaptainConfigProducer;
+import com.liveramp.captain.daemon.CaptainRequestConfig;
+import com.liveramp.captain.daemon.RequestUpdater;
+import com.liveramp.captain.daemon.SimpleCaptainConfig;
+import com.liveramp.captain.example.internals.MockAnalyticsService;
+import com.liveramp.captain.example.internals.MockDataScience;
+import com.liveramp.captain.example.internals.MockDb;
+import com.liveramp.captain.example.internals.MockIngestionService;
+import com.liveramp.captain.example.internals.MockRequest;
+import com.liveramp.captain.handle_persistor.HandlePersistor;
+import com.liveramp.captain.manifest.DefaultManifestImpl;
+import com.liveramp.captain.manifest.Manifest;
+import com.liveramp.captain.manifest_manager.ManifestManager;
+import com.liveramp.captain.manifest_manager.MultiAppManifestManager;
+import com.liveramp.captain.notifier.CaptainNotifier;
+import com.liveramp.captain.request_context.RequestContext;
+import com.liveramp.captain.request_submitter.NoOpSubmitter;
+import com.liveramp.captain.request_submitter.RequestSubmitter;
+import com.liveramp.captain.status.CaptainStatus;
+import com.liveramp.captain.status_retriever.StatusRetriever;
+import com.liveramp.captain.step.CaptainStep;
+import com.liveramp.captain.waypoint.AsyncWaypoint;
+import com.liveramp.captain.waypoint.ControlFlowWaypoint;
+import com.liveramp.captain.waypoint.SyncWaypoint;
+import com.liveramp.commons.collections.map.MapBuilder;
+import com.liveramp.daemon_lib.Daemon;
+
+public class ExampleCaptainWorkflow {
+  public static final String APP1 = "App1";
+  public static final String APP2 = "App2";
+  public static final String STEP1 = "Step1";
+  public static final String STEP2 = "Step2";
+  private static final String STEP3 = "Step3";
+  private static final String STEP4 = "Step4";
+  public static final String STEP5 = "DONE";
+
+  private Logger LOG = LoggerFactory.getLogger(ExampleCaptainWorkflow.class);
+
+  /**
+   * For sake of example, let's say our team is responsible for running some job that performs analytics on a file.
+   * <p>
+   * Doing this requires 3 main steps:
+   * 1) Wait for ingestion of the file to complete by some other team (i.e. poll until we know the file is ready for us to use.
+   * 2) Kick off a map reduce job that's handled by some service.
+   * 3) When that completes, report the information from the map reduce job to some other team's service. In this example,
+   * we'll have a split pipeline, where depending on the file, we report the information to the Data Science Team 1's
+   * Service or to Data Science Team 2's Service.
```

Comment:
Was a bit confused reading this, but maybe add `{@link MockDataScience}` and other related links so that its clear to match it up with the code later

Reply:
i'm not super familiar with the `@link` annotation. i feel like it's only valid in a javadoc comment. i can add it the javadoc style comments in this class. there are a lot of classes that aren't though, so i'll have to skip there, unless there's something i'm missing.

Reply:
The link is nice cause if I open it IntelliJ, i think it'll auto resolve and I can actually click into it and it'll jump to the class 

Ultimately, when the javadocs get rendered, it'll actually create a link to the javadoc for that file.